File: 214/projects/08/ruby/temperatureTable.rb
# temperatureTable.rb tests class Temperature and its operations
# by making a table of Temperatures
#
# Duncan Van Keulen for CS214 Project08
# Date: 4/13/2020
####################################################

load "Temperature.rb"
include TemperatureModule

def temperatureTable

    # Create and enter the values for the temprature objects
    baseTemp = Temperature.new(0.0, 'F')
    limitTemp = Temperature.new(0.0, 'F')

    print "Enter the base temperature: "
    baseTemp = baseTemp.enterTemperature()

    print "Enter the limit temperature: "
    limitTemp = limitTemp.enterTemperature()

    print "Enter the step value: "
    stepValue = gets.chomp.to_f

    puts "\n"
    puts "Fahrenheit\tCelsius\t\tKelvin"

    # Make a new Temperature to be the stepping Temperature
    stepTemp = baseTemp

    while (stepTemp.lessThan(limitTemp))
        stepTemp.toFahrenheit().displayTemperature()
        stepTemp.toCelsius().displayTemperature()
        stepTemp.toKelvin().displayTemperature()
        puts "\n"
        # Increment stepTemps degree by the stepValue
        stepTemp = stepTemp.raiseTemperature(stepValue)
    end #while

end # temperatureTable

temperatureTable

File: 214/projects/08/clojure/src/TemperatureTable.clj
;; TemperatureTable.clj tests a Clojure Temperature type,
;;
;; Usage: clojure -m TemperatureTable
;;
;; Completed by: Duncan Van Keulen
;; Date: 4/12/2020
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns TemperatureTable)

;; Load the Temperature module
(load-file "src/Temperature.clj")

(defn -main [] 
	;; Get the baseTemp from the user
	(print "Enter the base temperature: ") (flush)
	(let [ baseTemp (enterTemperature) ]

		;; Get the limitTemp from the user
		(print "Enter the limit temperature: ") (flush)
		(let [ limitTemp (enterTemperature) ]

			;; Get the stepValue from the user
			(print "Enter the step value: ") (flush)
			(let [ stepValue (read) ]
				(newline) (println "Fahrenheit              Celsius                 Kelvin") (newline)
				
				(loop [i 0] 
					;; Loop while the baseTemp is less than or equal to the limitTemp
					(when (or (lessThan (raiseTemperature baseTemp i) limitTemp) 
								(equals (raiseTemperature baseTemp i) limitTemp))
						(displayTemperature (raiseTemperature (toFahrenheit baseTemp) i))
						(print "		")
						(displayTemperature (raiseTemperature (toCelsius baseTemp) i))
						(print "		")
						(displayTemperature (raiseTemperature (toKelvin baseTemp) i))
						(newline) (flush)
						;; Recurse (loop) with i + stepvalue
						(recur (+ i stepValue))
					);; end when
					
				);; end loop
				
			);; end inner let
		)
	)
)
